lib/shared/shared-asset-bundler.ts

```diff
@@ -0,0 +1,49 @@
+import { BundlingOutput, DockerImage, IAsset, aws_s3_assets } from "aws-cdk-lib";
+import { AssetApiDefinition } from "aws-cdk-lib/aws-apigateway";
+import { Code, S3Code } from "aws-cdk-lib/aws-lambda";
+import { Asset } from "aws-cdk-lib/aws-s3-assets";
+import { md5hash } from "aws-cdk-lib/core/lib/helpers-internal";
+import { Construct } from "constructs";
+import * as path from "path";
+
+export class SharedAssetBundler extends Construct {
+    private readonly sharedAssets: string[];
+    private readonly WORKING_PATH = '/asset-input/'
+    /**
+     * Instantiate a new SharedAssetBundler. You then invoke `bundleWithAsset(pathToAsset)` to 
+     * bundle your asset code with the common code.
+     * 
+     * For Lambda function handler assets, you can use `bundleWithLambdaAsset(pathToAsset)` as 
+     * a drop-in replacement for `lambda.Code.fromAsset()`
+     * 
+     * @param scope 
+     * @param id 
+     * @param commonFolders : array of common folders to bundle with your asset code
+     */
+    constructor(scope: Construct, id: string, sharedAssets: string[]) {
+        super(scope, id);
+        this.sharedAssets = sharedAssets;
+    }
+
+    bundleWithAsset(assetPath: string): Asset {
+        const asset = new aws_s3_assets.Asset(this, md5hash(assetPath).slice(0, 6), {
+            path: assetPath,
+            bundling: {
+                image: DockerImage.fromBuild(path.join(__dirname, 'alpine-zip')),
```

Comment:
Should we use the `runtime.bundlingImage` defined [here](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/5776d3356a8a0bf17ee96aff15f1ca140fad32ca/lib/shared/index.ts#L11) to ensure we are bundling  compatible with the runtime in use?

Reply:
On this stage we are bundling just python files. It should not depend on the runtime. But matbe it's better to use default image because we already have downloaded it at this point

Reply:
We are just zipping files, so it does not matter which image we use. I chose `alpine` because it is small and I do not want to have dependencies as I wanted the construct to be reusable outside of this project

Reply:
OK all good